src/app/DomainPermissionsPage.qml

```diff
@@ -0,0 +1,290 @@
+                        toDelete.push(selectedDomainSetting.domain)
+                    }
+                    for (var i = 0; i < toDelete.length; i++) {
+                        DomainPermissionsModel.removeEntry(toDelete[i])
+                    }
+                    domainPermissionsListView.ViewItems.selectedIndices = []
+                    selectMode = false
+                }
+            },
+            Action {
+                iconName: "edit"
+                visible: !selectMode
+                enabled: domainPermissionsListView.count > 0
+                onTriggered: {
+                    selectMode = true
+                }
+            },
+            Action {
+                iconName: "add"
+                visible: !selectMode
+
+                onTriggered: {
+                    var promptDialog = PopupUtils.open(Qt.resolvedUrl("PromptDialog.qml"), domainPermissionsPage);
+                    promptDialog.title = i18n.tr("Add domain")
+                    promptDialog.message = i18n.tr("Enter the name of the domain, e.g. example.com (subdomains will be removed).")
+                    promptDialog.accept.connect(function(text) {
+                        if (text !== "") {
+                            var domain = DomainPermissionsModel.getDomainWithoutSubdomain(UrlUtils.extractHost(text));
+                            if (DomainPermissionsModel.contains(domain)) {
+                                domainPermissionsItem.setDomainAsCurrentItem(domain);
+                            }
+                            else {
+                                DomainPermissionsModel.insertEntry(domain, false);
+                                reload(domain);
+                            }
+                        }
+                    });
+                }
+            },
+            Action {
+                iconName: sortByLastRequested ? "clock" : "indicator-keyboard-Az"
+                visible: !selectMode
+                onTriggered: sortByLastRequested = !sortByLastRequested
+            }
+        ]
+
+        onBack: {
+            selectMode = false;
+            domainPermissionsItem.done();
+        }
+
+        ListView {
+            id: domainPermissionsListView
+            anchors.fill: parent
+            focus: true
+            model:  SortFilterModel {
+                model: DomainPermissionsModel
+                sort.order: sortByLastRequested ? Qt.DescendingOrder : Qt.AscendingOrder
+                sort.property: sortByLastRequested ? "lastRequested" : "domain"
+            }
+
+            ViewItems.selectMode: selectMode
+
+            delegate: ListItem {
+                id: item
+                readonly property bool isCurrentItem: item.ListView.isCurrentItem
+                readonly property string domain: model.domain
+                height: isCurrentItem ? layout.height : units.gu(5)
+                color: isCurrentItem ? theme.palette.selected.base : theme.palette.normal.background
```

Comment:
`base` is quite dark now. I suspect it is not the color you want.

Reply:
yes I think that was a trade-off where I tried to make it look acceptable with light and dark theme, corrections welcome : )